src/modules/emote_menu/search.js

```diff
@@ -0,0 +1,18 @@
+import Fuse from 'fuse.js';
+import watcher from '../../watcher.js';
+import emotes from '../emotes/index.js';
+
+class EmoteSearch {
+  constructor() {
+    this.options = {
+      keys: ['code'],
+    };
+    this.search = new Fuse(emotes.getEmotes(), this.options);
+
+    watcher.on('channel.updated', () => {
+      this.search = new Fuse(emotes.getEmotes(), this.options);
```

Comment:
rather than reinstantiating again, is there a way to update it with a new dataset? maybe `createIndex`?

i'm also curious how performance on this is. would it be more performant to remove emotes from the old channel and add emotes from the new channel?